CMS-Enterprise/sbom-harbor
# feat(detail_analytic): develop an analytic to produce the detailed report for the SDL team

## Summary

Implementation of `CMS SDL Specification for Data produced by SBOM Harbor`, which is found here:  
https://coda.io/d/SBOM-Harbor_dyJDHWU-tTl/CMS-SDL-Specification-for-Data-produced-by-SBOM-Harbor_sujo8#_lu9CO

### Added

1. Platform Analytic tooling to develop Aggregation Pipelines
2. Service and Task to support the Platform tooling
3. Data structures to support the report
4. CLI code to support running the Analytic from the CLI
5. Storage Providers for both the File System and  S3 

## How to test

1. Check out the branch and run `cargo test` to verify the tests pass
2. Run `cargo build` to build the executable
3. Make sure MongoDB is running locally
4. Use the Snyk SBOM Provider to get SBOMs: `harbor-cli sbom  snyk --debug`
5. Use the Snyk Enrichment Provider to enrich the SBOMs: `harbor-cli enrich  snyk --debug`
6. Use the Analysis Provider to create the Reports: `harbor-cli analyze detail --debug`
7. Verify in `/tmp/harbor-debug/detail`, that there are SDL Reports.
